debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java

```diff
@@ -443,16 +443,14 @@ protected ChangeRecordEmitter getChangeRecordEmitter(SnapshotContext<P, O> snaps
             overriddenSelect = connectorConfig.getSnapshotSelectOverridesByTable().get(new TableId(null, tableId.schema(), tableId.table()));
         }
 
-        return overriddenSelect != null ? Optional.of(enhanceOverriddenSelect(snapshotContext, overriddenSelect, tableId)) : getSnapshotSelect(snapshotContext, tableId);
-    }
+        if (overriddenSelect != null) {
+            return Optional.of(enhanceOverriddenSelect(snapshotContext, overriddenSelect, tableId));
+        }
 
-    /**
-     * Returns a comma-separated list of columns to be used in the snapshot select.
-     *
-     * @param tableId the table to generate a query for
-     * @return comma-separated list of columns
-     */
-    protected abstract String getSnapshotSelectColumns(TableId tableId);
+        List<String> columns = getPreparedColumnNames(schema.tableFor(tableId));
```

Comment:
@gunnarmorling The refactor looks good. But one question regarding the schema used here. This change won't work for Postgres. Please see https://issues.redhat.com/browse/DBZ-3818.

Reply:
@ani-sha, I might be wrong, but can't we change the type of `RelationalSnapshotChangeEventSource.schema` to `RelationalDatabaseSchema` (which it should have been from the get go), get rid of the `RelationalSnapshotChangeEventSource` constructor which doesn't take a schema and also remove the `schema` field in the Postgres snapshot source?

Reply:
@gunnarmorling I have updated the suggested changes and I think it should work, but the Postgres CI is still running with exceptions. 

FYI, the MySQL failure is unrelated.